src/clincoded/static/components/provisional_classification/snapshots.js

```diff
@@ -193,7 +207,7 @@ class CurationSnapshots extends Component {
                             </dl>
                         </td>
                         <td className="approval-snapshot-buttons">
-                            {this.props.allowPublishButton && snapshot['@id'] === currentApprovedSnapshotID && resourceParent && !isApprovalActive ?
+                            {this.props.allowPublishButton && !this.props.isPublishEventActive && snapshot['@id'] === currentApprovedSnapshotID && !isApprovalActive ?
```

Comment:
Why not put all these conditions into the `renderPublishLink()` function which returns JSX or doesn't depending on the boolean value.

Reply:
The function is called from multiple locations, each with a different set of conditions.

Reply:
I can see that `this.props.allowPublishButton && !this.props.isPublishEventActive && !isApprovalActive` are common conditions in the 3 places where the function is called.

Reply:
I've moved the common conditions into `renderPublishLink()`.